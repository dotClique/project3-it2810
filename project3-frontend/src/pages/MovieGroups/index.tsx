import { SearchIcon } from "@heroicons/react/solid";
import {
  AccordionDetails,
  AccordionSummary,
  InputAdornment,
  Pagination,
  TextField,
  Typography,
} from "@mui/material";
import { useState } from "react";
import { useHistory } from "react-router";
import MovieGroupItem from "../../components/MovieGroupItem";
import PageContainer from "../../components/PageContainer";
import {
  GroupAccordion,
  GroupGrid,
  LogOutButton,
  MovieGroupFooter,
  MovieGroupsContainer,
  NewGroupButton,
} from "./styled";

export default function MovieGroupsPage() {
  const [expanded, setExpanded] = useState("allMovies");
  const history = useHistory();

  const handleChange = (panel: string) => () => {
    setExpanded(panel);
  };

  return (
    <PageContainer>
      <MovieGroupsContainer>
        <Typography gutterBottom variant={"h3"}>
          Movie Groups
        </Typography>

        <GroupAccordion expanded={expanded === "favorites"} onChange={handleChange("favorites")}>
          <AccordionSummary aria-controls="panel1a-content" id="panel1a-header">
            <Typography>Favorites</Typography>
          </AccordionSummary>
          <AccordionDetails>
            <GroupGrid>
              <MovieGroupItem title={"Marvel"} favorite />
            </GroupGrid>
          </AccordionDetails>
        </GroupAccordion>
        <GroupAccordion expanded={expanded === "allMovies"} onChange={handleChange("allMovies")}>
          <AccordionSummary aria-controls="panel2a-content" id="panel2a-header">
            <Typography>All groups</Typography>
          </AccordionSummary>
          <AccordionDetails sx={{ maxHeight: "sm" }}>
            <TextField
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <SearchIcon width={20} />
                  </InputAdornment>
                ),
              }}
              placeholder={"search for groups"}
              sx={{ width: "90%", marginBottom: 1 }}
            />
            <GroupGrid>
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Sitcoms"} />
              <MovieGroupItem title={"Action"} />
              <MovieGroupItem title={"Another"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Sitcoms"} />
              <MovieGroupItem title={"Action"} />
              <MovieGroupItem title={"Another"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Marvel"} />
              <MovieGroupItem title={"Marvel"} />
            </GroupGrid>
          </AccordionDetails>
        </GroupAccordion>
        <MovieGroupFooter>
          <NewGroupButton>Add new movie group</NewGroupButton>
          <LogOutButton color={"secondary"} onClick={() => history.push("/")}>
            Change Alias
          </LogOutButton>
          <Pagination count={100} color="primary" />
        </MovieGroupFooter>
      </MovieGroupsContainer>
    </PageContainer>
  );
}
