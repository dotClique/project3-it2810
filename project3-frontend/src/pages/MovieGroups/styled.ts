import { Accordion, Button, Paper } from "@mui/material";
import { styled } from "@mui/material/styles";

export const MovieGroupsContainer = styled(Paper)(({ theme }) => ({
  marginTop: theme.spacing(8),
  borderRadius: theme.shape.borderRadius,
  position: "relative",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  maxWidth: "100vw",
  width: theme.breakpoints.values.md,
  [theme.breakpoints.down("md")]: {
    maxHeight: "none",
    height: "inherit",
    marginTop: theme.spacing(0),
    padding: "20px 4px 0",
  },
  padding: "20px 5vw 0",
}));

export const NewGroupButton = styled(Button)(({ theme }) => ({
  color: theme.palette.primary.contrastText,
  backgroundColor: theme.palette.primary.main,
  "&:hover": {
    backgroundColor: theme.palette.primary.light,
  },
  width: "100%",
  height: 50,
  marginTop: theme.spacing(2),
}));

export const MovieGroupFooter = styled("div")(({ theme }) => ({
  display: "flex",
  flexDirection: "column",
  gap: theme.spacing(3),
  marginTop: "auto",
  marginBottom: theme.spacing(2),
}));

export const GroupGrid = styled("div")(({ theme }) => ({
  display: "grid",
  width: "100%",
  gridTemplateColumns: "1fr 1fr",
  [theme.breakpoints.down("md")]: {
    gridTemplateColumns: "1fr",
  },
  overflow: "auto",
  gap: theme.spacing(1),
  padding: theme.spacing(1),
  "& > *": {
    minHeight: "80px",
  },
}));

export const GroupAccordion = styled(Accordion)(({ theme }) => ({
  width: "100%",
  border: `1px solid ${theme.palette.divider}`,
  boxShadow: "none",
  "&:hover": {
    filter: "brightness(90%)",
  },
  "&.Mui-expanded": {
    filter: "brightness(100%)!important",
  },
}));

export const LogOutButton = styled(Button)(({ theme }) => ({
  [theme.breakpoints.down("md")]: {
    position: "static",
    width: "100%",
    height: 50,
  },
  position: "absolute",
  left: theme.spacing(2),
  top: theme.spacing(2),
  color: theme.palette.secondary.contrastText,
  backgroundColor: theme.palette.secondary.main,
  "&:hover": {
    backgroundColor: theme.palette.secondary.light,
  },
}));
