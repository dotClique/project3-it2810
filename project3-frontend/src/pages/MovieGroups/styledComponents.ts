import { styled } from "@mui/system";
import { Button, Paper, Accordion } from "@mui/material";

export const MovieGroupsContainer = styled(Paper)(({ theme }) => ({
  marginTop: theme.spacing(8),
  borderRadius: theme.shape.borderRadius,
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  maxWidth: "100vw",
  width: theme.breakpoints.values.md,
  [theme.breakpoints.down("md")]: {
    maxHeight: "none",
    height: "inherit",
    marginTop: theme.spacing(0),
    padding: "20px 4px 0",
  },
  padding: "20px 5vw 0",
}));

export const NewGroupButton = styled(Button)(({ theme }) => ({
  color: theme.palette.primary.contrastText,
  backgroundColor: theme.palette.primary.main,
  "&:hover": {
    backgroundColor: theme.palette.primary.light,
  },
  width: "100%",
  height: 50,
  marginTop: theme.spacing(2),
}));

export const MovieGroupFooter = styled("div")(({ theme }) => ({
  display: "flex",
  flexDirection: "column",
  gap: theme.spacing(3),
  marginTop: "auto",
  marginBottom: theme.spacing(2),
}));

export const GroupGrid = styled("div")(({ theme }) => ({
  display: "grid",
  width: "100%",
  gridTemplateColumns: "1fr 1fr",
  [theme.breakpoints.down("md")]: {
    gridTemplateColumns: "1fr",
  },
  overflow: "auto",
  gap: theme.spacing(1),
  padding: theme.spacing(1),
  "& > *": {
    minHeight: "80px",
  },
}));

export const GroupAccordion = styled(Accordion)(({ theme }) => ({
  width: "100%",
  border: `1px solid ${theme.palette.divider}`,
  boxShadow: "none",
  "&:hover": {
    filter: "brightness(90%)",
  },
  "&.Mui-expanded": {
    filter: "brightness(100%)!important",
  },
}));
