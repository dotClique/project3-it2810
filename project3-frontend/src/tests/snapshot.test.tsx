import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import MovieGroups from "../components/MovieGroup";

Enzyme.configure({ adapter: new Adapter() });

describe("Snapshot tests", () => {
  // Snapshotting some components that many other components rely on and therefore should not be changed.
  test("Snapshotting MovieGroups", () => {
    const result = toJson(shallow(<MovieGroups />));
    expect(result).toMatchSnapshot();
  });
});
