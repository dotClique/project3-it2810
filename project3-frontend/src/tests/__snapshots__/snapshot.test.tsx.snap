// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Snapshotting MovieGroups 1`] = `
<Styled(ForwardRef(Card))>
  <Styled(ForwardRef(Link))
    color="inherit"
    href="/group/undefined"
    underline="none"
  >
    <Styled(ForwardRef(CardContent))>
      <Styled(ForwardRef(Typography))
        noWrap={true}
        variant="h5"
      />
    </Styled(ForwardRef(CardContent))>
  </Styled(ForwardRef(Link))>
  <FavoriteIcon
    isFilled={false}
    width={30}
  />
</Styled(ForwardRef(Card))>
`;
